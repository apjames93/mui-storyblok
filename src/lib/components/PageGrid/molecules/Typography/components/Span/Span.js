import React from 'react';
import PropTypes from 'prop-types';
import { Typography as MuiTypography } from '@material-ui/core';
import Storyblok from 'lib/utils/Storyblok';
import {
  dimensionProp,
  muiStringProp,
} from 'lib/utils/customProps';
import { useInView } from 'react-intersection-observer';

/**
 * Typography
 */


export const Span = ({
  align,
  color,
  height,
  width,
  rootClass,
  content,
  variant,
  key,
  storyblokClass,
  dataBlokC,
  dataBlokUid,
  transitionClass,
}) => {
  const { ref, inView } = useInView({ threshold: 0, triggerOnce: true });
  const defaultStyles = {
    height,
    width,
  };
  const styles = Storyblok.arrayToMuiStyles(rootClass, defaultStyles);

  return (
    <MuiTypography
      component="span"
      key={key}
      className={`${styles.root} ${storyblokClass} ${inView && transitionClass}`}
      variant={variant}
      align={align}
      color={color}
      data-blok-c={dataBlokC}
      data-blok-uid={dataBlokUid}
      inView={inView}
      ref={ref}
      style={{ opacity: inView ? 1 : 0 }}
    >
      {` ${content} `}
    </MuiTypography>
  );
};

export default Span;

Span.propTypes = {
  /**
   * storyblok multiselect of css classes
   * Override or extend the styles applied to the component
   * */
  rootClass: PropTypes.arrayOf(PropTypes.string),
  /**
   * mui prop: 'inherit', 'left', 'center', 'right', 'justify'
   * Set the text-align on the component.
   * */
  align(props, propName, componentName) {
    const validProps = ['inherit', 'left', 'center', 'right', 'justify'];
    return muiStringProp(props, propName, componentName, validProps);
  },
  /**
   * mui prop: 'initial', 'inherit', 'primary', 'secondary', 'textPrimary', 'textSecondary', 'error'
   * The color of the component. It supports those theme colors that make sense for this component.
   * */
  color(props, propName, componentName) {
    const validProps = ['initial', 'inherit', 'primary', 'secondary', 'textPrimary', 'textSecondary', 'error'];
    return muiStringProp(props, propName, componentName, validProps);
  },
  /** hight of the Typography element */
  height(props, propName, componentName) {
    return dimensionProp(props, propName, componentName);
  },
  /** width of the Typography element */
  width(props, propName, componentName) {
    return dimensionProp(props, propName, componentName);
  },

  content: PropTypes.string.isRequired,
  transitionClass: PropTypes.string,
};

Span.defaultProps = {
  width: '100%',
  height: '100%',
  align: 'inherit',
  color: 'initial',
  rootClass: [],
  transitionClass: '',
};
