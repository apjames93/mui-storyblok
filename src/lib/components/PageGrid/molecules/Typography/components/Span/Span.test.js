import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { Span } from './Span';

jest.mock('react-intersection-observer', () => ({
  useInView: () => ({ inView: true, ref: {} }),
}));

function setup(external = true) {
  const props = {
    content: 'wooo',
    external,
  };
  const comp = mount(<Span {...props} />);

  return { comp, props };
}

describe('<Span />', () => {
  it('renders Span', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create(<Span {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
