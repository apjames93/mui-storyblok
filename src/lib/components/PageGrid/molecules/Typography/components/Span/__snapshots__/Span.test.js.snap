// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Span /> snapshot 1`] = `
<span
  className="MuiTypography-root makeStyles-root-3   MuiTypography-body1"
  data-blok-c=""
  data-blok-uid=""
  inView={true}
  style={
    Object {
      "opacity": 1,
    }
  }
>
   wooo 
</span>
`;
